Sources/Stores/NetworkPersistableStore.swift

```diff
@@ -54,45 +53,46 @@ where Network.Remote == Data, Network.Request == URLRequest, Network.Response ==
-            failure: { [weak self] networkError in
-                // Check if it's cancelled
-                guard let strongSelf = self, cancelable.isCancelled == false
-                else { return completion(.failure(.network(networkError))) }
-
-                // try to fetch data from the Persistence as a fallback
-                strongSelf.persistenceFetch(
-                    resource,
-                    cacheHit: { [weak self] payload in
-                        self?.process(payload,
-                                      fromCache: true,
-                                      resource: resource,
-                                      cancelable: cancelable,
-                                      completion: completion) ?? completion(.failure(.cancelled(nil)))
-                    },
-                    cacheMiss: { completion(.failure(.network(networkError))) },
-                    failure: { completion(.failure(.multiple([networkError, $0]))) })
-        })
-
-        return cancelable
+        -> Cancelable
+        where R: NetworkResource & PersistableResource & RetryableResource,
+        R.Remote == Remote, R.Request == Request, R.Response == Response {
+
+            let cancelable = CancelableBag()
+
+            cancelable += networkFetch(
+                resource,
+                success: { [weak self ] payload in
+                    self?.process(payload,
+                                  fromCache: false,
+                                  resource: resource,
+                                  cancelable: cancelable,
+                                  completion: completion) ?? completion(.failure(.cancelled(nil)))
+                },
+                cancelled: { completion(.failure(.cancelled($0))) },
+                failure: { [weak self] networkError in
+                    // Check if it's cancelled
+                    guard let strongSelf = self, cancelable.isCancelled == false
+                        else { return completion(.failure(.network(networkError))) }
+
+                    // try to fetch data from the Persistence as a fallback
+                    strongSelf.persistenceFetch(
+                        resource,
+                        cacheHit: { [weak self] payload in
+                            self?.process(payload,
+                                          fromCache: true,
+                                          resource: resource,
+                                          cancelable: cancelable,
+                                          completion: completion) ?? completion(.failure(.cancelled(nil)))
+                        },
+                        cacheMiss: { completion(.failure(.network(networkError))) },
+                        failure: { completion(.failure(.multiple([networkError, $0]))) })
+                })
+
+            return cancelable
     }
 
     @discardableResult
+    // swiftlint:disable:next function_body_length
```

Comment:
does this have to be between `@discardableResult` and `private func`? 🤢

Reply:
Yeah it needs to be either in the line icmmeadetly before or after the line causing the problem. I changed this to the line after, but I don't think it got any better.